lib/platform/linux-i2c.c

```diff
@@ -367,24 +368,49 @@ static uint8_t i2c_gas_write_status_get(struct switchtec_dev *dev,
 	return -1;
 }
 
+static int i2c_gas_write_status_wait(struct switchtec_dev *dev,
+				     uint8_t tag)
+{
+	int status_retry_count = 0;
+	uint8_t status;
+
+	do {
+		status = i2c_gas_write_status_get(dev, tag);
+		if (status == 0 || status == GAS_TWI_MRPC_ERR)
+			break;
```

Comment:
Hold on a second... I just noticed there's already a retry loop inside i2c_gas_write_status_get() so we have three levels of retry loops... One must be redundant. Maybe we just need to catch another error inside that retry loop...

In any case, you could return here instead of breaking and deciding whether to return an error, but you'd also probably want to return an error in the case of GAS_TWI_MRPC_ERR....